app/models/service_retire_request.rb

```diff
@@ -1,6 +1,7 @@
 class ServiceRetireRequest < MiqRetireRequest
   TASK_DESCRIPTION  = 'Service Retire'.freeze
   SOURCE_CLASS_NAME = 'Service'.freeze
+  ACTIVE_STATES     = %w(retired) + base_class::ACTIVE_STATES
```

Comment:
Not for this PR, but "retired" doesn't sound like an active state, a better name for the validation to use might be VALID_STATES.

Reply:
Haha, I told her about this yesterday.

Reply:
@d-m-u @bdunne I know it looks odd, and we can rename it in the future, but retired is an active state.
Reconfigured and migrated are active states as well. 

Most of our lifecycle state machines have multiple states for each of 3 phases:
1. Pre-processing
2. Execution/Check Execution <---- the state would be provisioned/retired/reconfigured/migrated here
3. Post-processing <----- It still has to process these states before it can be finished.
The state is "finished" at the end of the state machine.